File: CalculatorBackend/src/computer/AddSubtractProcessor.java
package computer;

import java.util.Arrays;

import ifc.IfcMathExpression;

public class AddSubtractProcessor implements IfcMathExpression {

	private final IfcMathExpression decoratedExpression;

    public AddSubtractProcessor(IfcMathExpression decoratedExpression) {
        this.decoratedExpression = decoratedExpression;
    }

    @Override
    public String compute(String term) {
    	
    	term = term.replace("+-", "-").replace("--", "+");
    	
        term = decoratedExpression.compute(term); // Zuerst * und / berechnen

        if (!(term.contains("+") || term.contains("-"))) {
            return term;
        }

        //String regex = "(?<=[+*/])-|(?=[+*/])|(?<=\\d)-(?=\\d)|(?<=\\()|(?=\\))";
        //String[] numbers = term.split(regex);
        String[] numbers = term.split("[+\\-*/]");

        
        numbers = Arrays.stream(numbers)
                .filter(s -> !s.isEmpty()) // Entfernt leere Strings
                .toArray(String[]::new);

        
        String[] operators = term.replaceAll("[0-9.]", "").split("");

        double result = Double.parseDouble(numbers[0]);
        for (int i = 1; i < numbers.length; i++) {
            if (operators[i - 1].equals("+")) {
                result += Double.parseDouble(numbers[i]);
            } else if (operators[i - 1].equals("-")) {
                result -= Double.parseDouble(numbers[i]);
            }
        }

        return Double.toString(result);
    }
}

File: computer/AddSubtractProcessor.cs
using System;
using System.Linq;
using CalculatorBackend.ifc;

public class AddSubtractProcessor : IfcMathExpression
{
    private readonly IfcMathExpression _decoratedExpression;

    public AddSubtractProcessor(IfcMathExpression decoratedExpression)
    {
        _decoratedExpression = decoratedExpression;
    }

    public string Compute(string term)
    {
        term = term.Replace("+-", "-").Replace("--", "+");
        term = _decoratedExpression.Compute(term); // Zuerst * und / berechnen

        if (!(term.Contains("+") || term.Contains("-")))
        {
            return term;
        }

        string[] numbers = term.Split(new char[] { '+', '-', '*', '/' }, StringSplitOptions.RemoveEmptyEntries);
        string[] operators = term.Where(c => "+-*/".Contains(c)).Select(c => c.ToString()).ToArray();

        double result = double.Parse(numbers[0]);
        for (int i = 1; i < numbers.Length; i++)
        {
            if (operators[i - 1] == "+")
            {
                result += double.Parse(numbers[i]);
            }
            else if (operators[i - 1] == "-")
            {
                result -= double.Parse(numbers[i]);
            }
        }

        return result.ToString();
    }
}
